### PR TITLE
removing unused OCAMLRUNPARAM parameters from the code and from the manual

### PR DESCRIPTION
**Pull request by sabine**
Relating to ocaml-multicore/ocaml-multicore#794

backtrace_enabled: in use
cleanup_on_exit: in use
eventlog_enabled: in use
init_fiber_wsz: in use
init_heap_wsz: UNUSED
init_heap_chunk_sz: UNUSED
init_max_stack_wsz: in use 
init_custom_major_ratio: in use
init_custom_minor_ratio: in use
init_custom_minor_max_bsz: in use
init_percent_free: in use
init_max_percent_free: UNUSED
parser_trace: in use
init_minor_heap_wsz: in use
trace_level: in use
verb_gc: in use
verify_heap: in use
caml_runtime_warnings: in use

I found some options that were already commented out in 750e2123076d99bc5d04b323d8aa976a832eb74c and that have no implementation associated with them:
- caml_set_allocation_policy - has a manual entry
- caml_use_huge_pages - has a manual entry
- caml_init_major_window - has a manual entry

I noticed there are discrepancies between the names in the code and in the manual:
For example, `init_max_percent_free` has a manual entry `\item[O] ("max_overhead")`. I assume this is intentional.

- [x] remove the code associated with the unused parameters
- [x] update the manual chapter 13 `runtime.etex` to remove all unused parameters, including the ones that already had no implementation
- [x] remove `init_fiber_wsz` option

To discuss: do any of the removed options need to be reimplemented for multicore?

### CODE REVIEW

**Comment by kayceesrk** [comment]
Thanks for the contribution. The changes look good to me. 

Since we are on the topic of removal of unused options, it may be useful to also remove the `init_fiber_wsz` option as well. This was added by the multicore team to evaluate the performance of the different initial size choices for the fiber, and is not really used in any of our programs. The default size of 32 words that we currently have is good enough in our experiments.

**Comment by sabine** [comment]
I checked for consistency between the manual entries for OCAMLRUNPARAM and the implementation and found that two parameters don't have a manual entry:

- `verify_heap`
- `eventlog_enabled`

What should be done about that?

**Comment by kayceesrk** [comment]
For `verify_heap`, it may be useful to document that it "runs an integrity check on the heap just after the completion of a major GC cycle". And leave it at that.  

For reference, the actual check is here: https://github.com/ocaml/ocaml/blob/001997e81342fd0d321fd877b73608150601e7d9/runtime/shared_heap.c#L656-L658. 

@Engil may comment about `eventlog_enabled`. Does eventlog work as expected or is there more work?

**Comment by abbysmal** [comment]
Eventlog should work as expected.
The manual entry for this specific option is actually to be found in the [instrumented runtime manual page](https://ocaml.org/manual/instrumented-runtime.html).

**Comment by kayceesrk** [comment]
@Engil Could you provide the documentation string that should be attached to the OCAMLRUNPARAM option "e"? See https://ocaml.org/manual/runtime.html#s:ocamlrun-options where other options are documented.

**Comment by abbysmal** [comment]
@sabine @kayceesrk My bad. I was misleading.
`OCAMLRUNPARAM=e` is not the way to enable eventlog as of the initial integration of the feature in OCaml 4.10.0.
The `e` option is a leftover of the early implementation found in Multicore.
Eventlog is enabled in trunk by using `OCAML_EVENTLOG_ENABLED`, not `OCAMLRUNPARAM`, so the `e` option can be dropped.

**Comment by kayceesrk** [comment]
Ok. @sabine please remove the `e` option. I think it is good to go after that.

**Comment by kayceesrk** [comment]
LGTM!

Could one of the core devs review this PR? CC @damiendoligez @gasche.

**Review by gasche — APPROVED** [review]
I'm happy to approve (and eventually merge) on @kayceesrk's behalf.

One question though: some of those settings correspond to control knobs for the GC behavior (initial size of the major heap and heap increment). Do we know that those knobs are not desirable for the new GC, or is it just that they would be helpful, but they have not been implemented yet?

If we would like to re-enable those tuning parameters later, it could make sense to open an issue to track this feature wish, and we might want to handle their removal slightly differently (or not: people can use the git history to find the deleted documentation).

**Comment by kayceesrk** [comment]
> One question though: some of those settings correspond to control knobs for the GC behavior (initial size of the major heap and heap increment). Do we know that those knobs are not desirable for the new GC, or is it just that they would be helpful, but they have not been implemented yet?

These parameters are not directly interpretable in the new GC as the heap structure is quite different. There may be other useful knobs in the new GC. @sadiqj and @ctk21 would know if there was.

**Comment by gasche** [comment]
Merged. Thanks @sabine for your contribution.

**Pull request merged**
